src/main/java/org/truffleruby/core/hash/HashLiteralNode.java

```diff
@@ -29,10 +30,18 @@ protected HashLiteralNode(RubyNode[] keyValues) {
         this.keyValues = keyValues;
     }
 
-    private static final boolean bigHashTypeIsCompactHash;
-    static {
-        String type = System.getProperty("BigHashStrategy");
-        bigHashTypeIsCompactHash = type != null && type.equalsIgnoreCase("compact");
+    // This singleton serves no other purpose than being able to call .getContext() (an instance method) from
+    static final class CreateBigHashLiteralNode extends RubyBaseNode {
+        private CreateBigHashLiteralNode() {
+        }
+
+        private static final CreateBigHashLiteralNode INSTANCE = new CreateBigHashLiteralNode();
+
+        private HashLiteralNode execute(RubyNode[] keyValues) {
+            return getContext().getOptions().BIG_HASH_STRATEGY_IS_BUCKETS
+                    ? new BucketsHashStore.GenericHashLiteralNode(keyValues)
+                    : new CompactHashStore.CompactHashLiteralNode(keyValues);
+        }
```

Comment:
This feels too hacky, could you instead pass RubyLanguage to `HashLiteralNode.create`, and make `BIG_HASH_STRATEGY_IS_BUCKETS` a language option? (see `LANGUAGE_OPTIONS:` in options.yml)